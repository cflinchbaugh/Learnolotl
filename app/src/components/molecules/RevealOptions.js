import React, { Component } from 'react';
import styled from 'styled-components';
import CancelPop from 'molecules/buttons/CancelPop';
import { connect } from 'react-redux';

const StyleWrapper = styled.div`
    .RevealOptions {
        margin: auto;
    }
    
    .RevealOptions >* {
        float: left;
        margin-right: 2em;
    }

    .RevealOptions >*:last-child {
        margin-right: 0;
    }

`;

class RevealOptions extends Component {    
    _processReveal(translationsArr) {
        return translationsArr.map(function(translationData, idx) {
            let reveal,
                buttonData = {
                    label: translationData.id,
                    onClickFunction: '',
                    showX: false
                },
                displayDataId = this.props.flashCardDisplay[0] ? this.props.flashCardDisplay[0].id: false;

            if (translationData.id === displayDataId) {
                buttonData.onClickFunction = (() => this.handleClick('none'));
                buttonData.showX = true
            } else {
                buttonData.onClickFunction = (() => this.handleClick(translationData.id));
            }
            
            reveal = (
                <CancelPop key={idx} {...buttonData}/>
            );
            
            return reveal;
        }, this);
    }
    
    render() {
        let reveal = this._processReveal(this.props.optionsArray);
        
        return (
            <StyleWrapper>
                <div className="RevealOptions clearfix">
                    {reveal}
                </div>
            </StyleWrapper>
        )
    }

    handleClick(e) {
        this.props.handleClickOption(e);
    }

}

const mapStateToProps = state => ({
    flashCardDisplay: state.cards.revealOption
});

export default connect(mapStateToProps, {  })(RevealOptions);