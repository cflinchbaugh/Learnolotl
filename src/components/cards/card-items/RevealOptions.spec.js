import React from 'react';
import {shallow} from 'enzyme';
import {RevealOptions} from './RevealOptions';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('RevealOptions', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const revealOptionsData = {
            optionsArray: []
        }

        const tree = renderer.render(
            <RevealOptions {...revealOptionsData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    it('renders translationData and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const revealOptionsData = {
            optionsArray: [
                {
                    id: 'English'
                }, {
                    id: 'Hiragana'
                }, {
                    id: 'Romaji'
                }
            ],
            flashCardDisplay: []
        }

        const tree = renderer.render(
            <RevealOptions {...revealOptionsData}/>
        );

        expect(tree).toMatchSnapshot();
    });
});