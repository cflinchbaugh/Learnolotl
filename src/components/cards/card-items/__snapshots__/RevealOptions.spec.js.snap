// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevealOptions renders and matches snapshot 1`] = `
<styled.div
  className="reveal-options-wrapper"
>
  <div
    className="RevealOptions clearfix"
  >
    
  </div>
</styled.div>
`;

exports[`RevealOptions renders translationData and matches snapshot 1`] = `
<styled.div
  className="reveal-options-wrapper"
>
  <div
    className="RevealOptions clearfix"
  >
    <CancelPopButton
      label="English"
      onClickFunction={[Function]}
      showX={false}
    />
    <CancelPopButton
      label="Hiragana"
      onClickFunction={[Function]}
      showX={false}
    />
    <CancelPopButton
      label="Romaji"
      onClickFunction={[Function]}
      showX={false}
    />
  </div>
</styled.div>
`;
